core/src/main/java/com/exadel/aem/toolkit/core/injectors/EToolboxListInjector.java

```diff
@@ -0,0 +1,174 @@
+        @Nonnull Type type,
+        @Nonnull AnnotatedElement element,
+        @Nonnull DisposalCallbackRegistry callbackRegistry) {
+
+        EToolboxList annotation = element.getDeclaredAnnotation(EToolboxList.class);
+        if (annotation == null) {
+            return null;
+        }
+
+        ResourceResolver resourceResolver = InjectorUtils.getResourceResolver(adaptable);
+        if (resourceResolver == null) {
+            return null;
+        }
+
+        if (InjectorUtils.isValidRawType(type, Collection.class)) {
+
+            return getTypeArgument(type, 0).equals(Object.class)
+                ? getObjectList(resourceResolver, annotation.value())
+                : ListHelper.getList(resourceResolver, annotation.value(), getTypeArgument(type, 0));
+
+        } else if (InjectorUtils.isValidRawType(type, Map.class)) {
+
+            if (!annotation.keyProperty().isEmpty()) {
+                return ListHelper.getMap(resourceResolver, annotation.value(), annotation.keyProperty(), getTypeArgument(type, 1));
+
+            } else if (getTypeArgument(type, 1).equals(String.class) || getTypeArgument(type, 1).equals(Object.class)) {
+                return ListHelper.getMap(resourceResolver, annotation.value());
+            }
+
+            return ListHelper.getMap(resourceResolver, annotation.value(), JcrConstants.JCR_TITLE, getTypeArgument(type, 1));
+
+        } else if (!(type instanceof ParameterizedType) && ((Class<?>) type).isArray()) {
+            List<?> modelList = ListHelper.getList(resourceResolver, annotation.value(), ((Class<?>) type).getComponentType());
+            List<Object> objectList = getObjectList(resourceResolver, annotation.value());
+
+            return ((Class<?>) type).getComponentType().equals(Object.class)
+                ? toArray(objectList, ((Class<?>) type).getComponentType())
+                : toArray(modelList, ((Class<?>) type).getComponentType());
+        }
+
+        LOG.debug(InjectorConstants.EXCEPTION_UNSUPPORTED_TYPE, type);
+        return null;
+    }
+
+    /**
+     * Retrieves a {@code List<Object>} collected from list entries under the provided path
+     * @param resourceResolver Sling {@code ResourceResolver} instance used to access the list
+     * @param path             The path to a list of items
+     * @return List of {@code Object}
+     */
+    private List<Object> getObjectList(ResourceResolver resourceResolver, String path) {
+
+        return StreamSupport.stream(Spliterators.spliteratorUnknownSize(
+            Objects.requireNonNull(Optional.of(resourceResolver)
+                    .map(resolver -> resolver.adaptTo(PageManager.class))
+                    .map(pageManager -> pageManager.getPage(path))
+                    .map(Page::getContentResource)
+                    .map(contentRes -> contentRes.getChild("list"))
+                    .orElse(null))
+                .listChildren(), Spliterator.ORDERED), false).collect(Collectors.toList());
+    }
```

Comment:
I had to add this method here because the ListHelper only has a method that returns a List <T>, and inside this method, the resource adapts to the given class, but if it's an Object class, then I get an empty collection. So I removed the step for mapping with the specified type and added this method.